### PR TITLE
`erlang`: Bump to 24.0.3

### PR DESCRIPTION
**Pull request by thunder-coding**

### CODE REVIEW

**Comment by thunder-coding** [comment]
Seems like I messed up `emulator/Makefile.in`

**Comment by thunder-coding** [comment]
It has been a week since I created PR, it would be very helpful for me if anyone from the `Termux` team helps me identify the root of build failure, Thanks in advance!

**Comment by suhan-paradkar** [comment]
> 2021-07-02T05:44:06.3153464Z  RANLIB	../lib/internal/aarch64-unknown-linux-android/liberts_internal_r.a
2021-07-02T05:44:06.3221036Z  GEN	obj/aarch64-unknown-linux-android/opt/MADE
2021-07-02T05:44:06.3229147Z make[7]: Leaving directory '/home/builder/.termux-build/erlang/src/erts/lib_src'
2021-07-02T05:44:06.3235640Z make[6]: Leaving directory '/home/builder/.termux-build/erlang/src/erts/lib_src'
2021-07-02T05:44:06.3249305Z  GEN	aarch64-unknown-linux-android/erl_db_insert_list.ycf.h
2021-07-02T05:44:06.3256710Z  GEN	aarch64-unknown-linux-android/erl_map.ycf.h
2021-07-02T05:44:06.3284966Z /bin/sh: 1: /home/builder/.termux-build/erlang/src/erts/lib_src/yielding_c_fun/bin/aarch64-unknown-linux-android/yielding_c_fun: Exec format error
2021-07-02T05:44:06.3291956Z  GEN	aarch64-unknown-linux-android/utils.ycf.h
2021-07-02T05:44:06.3326650Z /bin/sh: 1: /home/builder/.termux-build/erlang/src/erts/lib_src/yielding_c_fun/bin/aarch64-unknown-linux-android/yielding_c_fun: Exec format error
2021-07-02T05:44:06.3329201Z /bin/sh: 1: /home/builder/.termux-build/erlang/src/erts/lib_src/yielding_c_fun/bin/aarch64-unknown-linux-android/yielding_c_fun: Exec format error
2021-07-02T05:44:06.3416289Z make[5]: *** No rule to make target 'aarch64-unknown-linux-android/erl_db_insert_list.ycf.h', needed by 'aarch64-unknown-linux-android/opt/emu/GENERATED'.  Stop.
2021-07-02T05:44:06.3418679Z make[5]: Leaving directory '/home/builder/.termux-build/erlang/src/erts/emulator'
2021-07-02T05:44:06.3426563Z make[4]: *** [/home/builder/.termux-build/erlang/src/make/run_make.mk:35: opt] Error 2
2021-07-02T05:44:06.3428109Z make[4]: Leaving directory '/home/builder/.termux-build/erlang/src/erts/emulator'
2021-07-02T05:44:06.3436670Z make[3]: Leaving directory '/home/builder/.termux-build/erlang/src/erts'
2021-07-02T05:44:06.3437678Z make[3]: *** [Makefile:45: opt] Error 2
2021-07-02T05:44:06.3444980Z make[2]: *** [Makefile:54: emu] Error 2
2021-07-02T05:44:06.3446586Z make[2]: Leaving directory '/home/builder/.termux-build/erlang/src/erts'
2021-07-02T05:44:06.3454880Z make[1]: *** [Makefile:483: emulator] Error 2
2021-07-02T05:44:06.3455958Z make[1]: Leaving directory '/home/builder/.termux-build/erlang/src'
2021-07-02T05:44:06.3464219Z make: *** [Makefile:389: noboot] Error 2
2021-07-02T05:44:06.3565957Z ##[error]Process completed with exit code 2.

Well... looks like it tries to execute target binary in host... and mostly that's why you get Exec format error..

**Comment by thunder-coding** [comment]
So here's the culprit:

Makefile.in line 571
```
bootstrap_setup: check_recreate_primary_bootstrap bootstrap_setup_target $(BOOTSTRAP_ROOT)/bootstrap/bin/erl $(BOOTSTRAP_ROOT)/bootstrap/bin/erlc $(BOOTSTRAP_ROOT)/bootstrap/bin/escript $(BOOTSTRAP_ROOT)/bootstrap/bin/yielding_c_fun
```

**Comment by thunder-coding** [comment]
The exec format error was solved by passing appropriate `--host` to configure. See http://erlang.org/documentation/doc-5.7.5/doc/installation_guide/source/INSTALL-CROSS.html

**Comment by xeffyr** [comment]
You still have issues with cross-compiling:
```
/home/builder/.termux-build/_cache/android-r21d-api-24-v4/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: error: /home/builder/.termux-build/erlang/src/lib/erl_interface/obj/aarch64-unknown-linux-android/libei.a(ei_compat.o): incompatible target
/home/builder/.termux-build/_cache/android-r21d-api-24-v4/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: error: /home/builder/.termux-build/erlang/src/lib/erl_interface/obj/aarch64-unknown-linux-android/libei.a(ei_init.o): incompatible target
```

File: packages/erlang/build.sh [hunk 1]
```diff
@@ -20,8 +20,8 @@ termux_step_post_get_source() {
 
 termux_step_host_build() {
 	cd $TERMUX_PKG_SRCDIR
-	./configure --enable-bootstrap-only
-	make -j "$TERMUX_MAKE_PROCESSES"
+	./configure --enable-bootstrap-only --host=$TERMUX_HOST_PLATFORM --build=$TERMUX_BUILD_TUPLE
```
Comment (xeffyr) [inline packages/erlang/build.sh:23]: Why you are doing this for **host** build? These flags should be passed only for cross-compilation steb, but in termux_step_host_build you need to compile erlang for Ubuntu x86_64.

**Review by xeffyr — CHANGES_REQUESTED** [review]

File: packages/erlang/build.sh (hunk 1 above)
Comment (thunder-coding) [reply packages/erlang/build.sh:23]: Probably we don't need to host build Erlang. https://git.alpinelinux.org/aports/tree/community/erlang/APKBUILD

See Alpine Linux APKBUILD file. Passing appropriate `--host` and `--build` should do

**Comment by thunder-coding** [comment]
Seems like Erlang is compiling for the wrong architecture
https://stackoverflow.com/questions/50518083/cross-compiling-a-c-program-for-android/50518522

**Pull request closed without merging**